File: course-service/src/main/java/com/quyennv/datn/courseservice/core/domain/valueobject/Resource.java
package com.quyennv.datn.courseservice.core.domain.valueobject;

import com.quyennv.datn.courseservice.core.domain.entities.Identity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Resource {
    private Identity id;
    private String name;
    private String type;
    private String url;
    private String videoDuration;
    private String storeProvider;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime deletedAt;
}

File: file-service/core/entities/resource.go
package entities

import (
	"time"

	"file-service/core/common"
	"github.com/google/uuid"
)

type Resource struct {
	Id            uuid.UUID  `json:"id"`
	Name          string     `json:"name"`
	StoreProvider string     `json:"storeProvider"`
	Type          string     `json:"type"`
	Url           string     `json:"url"`
	VideoDuration int        `json:"videoDuration"`
	Resolution    string     `json:"resolution"`
	CreatedAt     *time.Time `json:"createdAt"`
	UpdatedAt     *time.Time `json:"updatedAt"`
	DeletedAt     *time.Time `json:"deletedAt"`
}

func ErrCannotSaveFile(err error) *common.AppError {
	return common.NewCustomError(err, "cannot save file", "ErrCannotSaveFile")
}
